common4j/src/main/com/microsoft/identity/common/java/controllers/BaseController.java

```diff
@@ -857,6 +857,7 @@ protected TokenResult performSilentTokenRequest(
             ((MicrosoftTokenRequest) refreshTokenRequest).setClaims(parameters.getClaimsRequestJson());
             ((MicrosoftTokenRequest) refreshTokenRequest).setClientAppName(parameters.getApplicationName());
             ((MicrosoftTokenRequest) refreshTokenRequest).setClientAppVersion(parameters.getApplicationVersion());
+            ((MicrosoftTokenRequest) refreshTokenRequest).setCorrelationId(UUID.fromString(parameters.getCorrelationId()));
```

Comment:
@shahzaibj can you lean in here? @SaurabhMSFT  can provide context if needed.

Reply:
The correlation id was set in the old AcquireTokenNoFixedScope Command.  I also see that the correlation id is being set in different subclasses of TokenRequest. This sets the correlation id we have from from account state class for new token retrieval or a token refresh.

Reply:
> This sets the correlation id we have from from account state class for new token retrieval or a token refresh.

True, but it does more than that. It sets the correlation ID for _all_ MicrosoftTokenRequests, including those going through browser-delegated flows. MSAL team might have different requirements. I want to avoid accidental side effects, so let's have it reviewed by their team.

Reply:
This discussion has been moved to PR https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2413/files